Library/Formula/gnuradio.rb

```diff
@@ -0,0 +1,230 @@
+  end
+  # sphinx ends here
+
+  resource "cppzmq" do
+    url "https://github.com/zeromq/cppzmq/raw/34c8e4395c94d34a89bbeaaf2b8f9c94a8293c84/zmq.hpp"
+    sha256 "6cdd9d920f4a0f9f3a3257541321bef8369d9735c88b84ba8ae0e461f53e476c"
+  end
+
+  option "with-python", "Build python support"
+  option "with-sphinx", "Build sphinx document"
+  option "with-wavelet", "Build gr-wavelet"
+  option "with-wx", "Build wx gui (requires --with-python)"
+  option "with-qt", "Build qt gui (requires --with-python)"
+  option "with-zeromq", "BUild gr-zeromq"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "swig" => :build if build.with? "python"
+
+  depends_on "boost"
+  depends_on "cppunit"
+  depends_on "fftw"
+  depends_on "python"
+  depends_on "homebrew/python/numpy"
+  depends_on "gsl" if build.with? "wavelet"
+  depends_on "qt" if build.with? "python"
+  depends_on "qwt" if build.with? "python"
+  depends_on "pyqt" if build.with? "python"
+  depends_on "wxpython" if build.with? "wx"
+  depends_on "sdl" => :optional
+  depends_on "uhd" => :optional
+  depends_on "doxygen" => :optional
+
+  def install
+    if (build.with? "wx") && ! (build.with? "python")
+      raise "--with-wx depends on --with-python"
+    end
+
+    if (build.with? "qt") && ! (build.with? "python")
+      raise "--with-qt depends on --with-python"
+    end
+
+    ENV.append "LDFLAGS", "-undefined dynamic_lookup"
+
+    ENV["CHEETAH_INSTALL_WITHOUT_SETUPTOOLS"] = "1"
+    ENV.prepend_create_path "PYTHONPATH", libexec+"lib/python2.7/site-packages"
+    install_args = ["setup.py", "install", "--prefix=#{libexec}"]
+
+    resource("Markdown").stage { system "python", *install_args }
+    resource("Cheetah").stage { system "python", *install_args }
+    resource("lxml").stage { system "python", *install_args }
+
+    if build.with? "sphinx"
+      resource("sphinx").stage { system "python", *install_args }
+      resource("sphinx_rtd_theme").stage { system "python", *install_args }
+      resource("alabaster").stage { system "python", *install_args }
+      resource("babel").stage { system "python", *install_args }
+      resource("docutils").stage { system "python", *install_args }
+      resource("pygments").stage { system "python", *install_args }
+      resource("jinja2").stage { system "python", *install_args }
+      resource("markupsafe").stage { system "python", *install_args }
+      resource("snowballstemmer").stage { system "python", *install_args }
+      resource("six").stage { system "python", *install_args }
+      resource("pytz").stage { system "python", *install_args }
+    end
+
+    if build.with? "zeromq"
+      resource("cppzmq").stage "#{include}"
+    end
+
+    ENV.prepend_path "PATH", (libexec+"bin")
+
+    args = std_cmake_args
+    args << "-DENABLE_DEFAULT=OFF"
+    args << "-DENABLE_PYTHON=ON" if build.with? "python"
+    args << "-DENABLE_GR_ANALOG=ON"
```

Comment:
If some of these are enabled by default, you don't need to declare them all `ON`. Only do that when we want to change the default setting.